contributors/design-proposals/kubectl-create-configmap-env-file.md

```diff
@@ -0,0 +1,65 @@
+# Kubectl create configmap --env-file 
+
+## Goals
+
+`kubectl create configmap` allows a user to create configmaps via the command
+line. There should be easier ways to create a `ConfigMap`. One such way is to
+allow a Docker environment file (.env) to populate a `ConfigMap`.
+
+## Design
+
+The `create configmap` subcommand would add a new option called
+`--from-env-file`. The option will accept a single file. The option may not be
+used in conjunction with `--from-file` or `--from-literal-value`.
+
+### Environment file specification
+
+An environment file consists of lines to be in VAR=VAL format. Lines beginning
+with # (i.e. comments) are ignored, as are blank lines. Any whitespace in
+front of the VAR is removed. VAR must be a valid C_IDENTIFIER.  If the line
+consists of just VAR, then the VAL will be given a value from the current
+environment.
+
+[Docker's environment file processing](https://github.com/docker/docker/blob/master/runconfig/opts/envfile.go)
+
+## Examples
+
+```
+$ ls 
+game.env
+ui.env
+
+$ cat game.env
+enemies=aliens
+lives=3
+enemies.cheat=true
```

Comment:
It was copy and paste error from the proposal I stole this from.
It is now trimmed and corrected.